File: ISL.ReIdentification.WebApps.Portals/ISL.ReIdentification.Portals.Server.Tests.Acceptance/Models/CsvIdentificationRequests/CsvIdentificationRequest.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;

namespace ISL.ReIdentification.Portals.Server.Tests.Acceptance.Models.CsvIdentificationRequests
{
    public class CsvIdentificationRequest
    {
        public Guid Id { get; set; }
        public Guid RequesterEntraUserId { get; set; } = Guid.Empty;
        public string RequesterFirstName { get; set; } = string.Empty;
        public string RequesterLastName { get; set; } = string.Empty;
        public string RequesterDisplayName { get; set; } = string.Empty;
        public string RequesterEmail { get; set; } = string.Empty;
        public string RequesterJobTitle { get; set; } = string.Empty;
        public Guid RecipientEntraUserId { get; set; } = Guid.Empty;
        public string RecipientFirstName { get; set; } = string.Empty;
        public string RecipientLastName { get; set; } = string.Empty;
        public string RecipientDisplayName { get; set; } = string.Empty;
        public string RecipientEmail { get; set; } = string.Empty;
        public string RecipientJobTitle { get; set; } = string.Empty;
        public string Reason { get; set; } = string.Empty;
        public string Organisation { get; set; } = string.Empty;
        public string Filepath { get; set; } = string.Empty;
        public byte[] Data { get; set; } = Array.Empty<byte>();
        public string Sha256Hash { get; set; } = string.Empty;
        public int IdentifierColumnIndex { get; set; } = 0;
        public bool HasHeaderRecord { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTimeOffset CreatedDate { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
    }
}

File: ISL.ReIdentification.WebApps.Configurations/ISL.ReIdentification.Configurations.Client/src/models/csvIdentificationRequest/csvIdentificationRequest.ts
export class CsvIdentificationRequest {
    public id: string;
    public requesterFirstName: string;
    public requesterLastName: string;
    public requesterEmail: string;
    public recipientFirstName: string;
    public recipientLastName: string;
    public reason: string;
    public purpose: string;
    public organisation: string;
    public data: Uint8Array;
    public sha256Hash: string;
    public identifierColumn: string;
    public filepath: string;
    public createdBy?: string;
    public createdDate?: Date | undefined;
    public updatedBy?: string;
    public updatedDate?: Date | undefined;

    constructor(csvIdentificationRequest: CsvIdentificationRequest) {
        this.id = csvIdentificationRequest.id ? csvIdentificationRequest.id : "";
        this.requesterFirstName = csvIdentificationRequest.requesterFirstName || "";
        this.requesterLastName = csvIdentificationRequest.requesterLastName || "";
        this.requesterEmail = csvIdentificationRequest.requesterEmail || "";
        this.recipientFirstName = csvIdentificationRequest.recipientFirstName || "";
        this.recipientLastName = csvIdentificationRequest.recipientLastName || "";
        this.reason = csvIdentificationRequest.reason || "";
        this.purpose = csvIdentificationRequest.purpose || "";
        this.organisation = csvIdentificationRequest.organisation || "";
        this.data = csvIdentificationRequest.data || 0;
        this.sha256Hash = csvIdentificationRequest.sha256Hash || "";
        this.identifierColumn = csvIdentificationRequest.identifierColumn || ""; 
        this.filepath = csvIdentificationRequest.filepath || ""; 
        this.createdDate = csvIdentificationRequest.createdDate ? new Date(csvIdentificationRequest.createdDate) : undefined;
        this.createdBy = csvIdentificationRequest.createdBy || "";
        this.updatedDate = csvIdentificationRequest.updatedDate ? new Date(csvIdentificationRequest.updatedDate) : undefined;
        this.updatedBy = csvIdentificationRequest.updatedBy || "";
    }
}